File: src/backend/app/api/v1/endpoints/health.py
import logging
from datetime import datetime, timezone
from fastapi import APIRouter, status
from fastapi.responses import JSONResponse
from app.core.database import check_database_connection
from app.core.config import settings

logger = logging.getLogger("sentinelai.api.health")

router = APIRouter()

@router.get(
    "/health",
    summary="SentinelAI System & Database Health",
    description="Returns operational status of the backend API and the connected PostgreSQL database."
)
def api_v1_health():
    """
    Check the operational status of the SentinelAI v1 API and PostgreSQL database.
    """
    db_ok = check_database_connection()
    timestamp = datetime.now(timezone.utc).isoformat()

    if db_ok:
        return {
            "status": "healthy",
            "service": settings.APP_NAME,
            "version": "1.0.0",
            "database": "connected",
            "database_type": "PostgreSQL",
            "timestamp": timestamp
        }
    else:
        logger.error("Database connection check failed during /api/v1/health call.")
        return JSONResponse(
            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
            content={
                "status": "unhealthy",
                "service": settings.APP_NAME,
                "version": "1.0.0",
                "database": "disconnected",
                "database_type": "PostgreSQL",
                "timestamp": timestamp
            }
        )

File: src/frontend/src/api/health.js
import apiClient from './client';
import { API_CONFIG } from '../config/api.config';

/**
 * Checks system and PostgreSQL database health via /api/v1/health.
 * @returns {Promise<{ status: string, service: string, version: string, database: string, database_type: string, timestamp: string }>}
 */
export async function getSystemHealth() {
  return await apiClient(API_CONFIG.ENDPOINTS.HEALTH);
}

/**
 * Checks legacy root /health endpoint.
 */
export async function getRootHealth() {
  return await apiClient(API_CONFIG.ENDPOINTS.ROOT_HEALTH);
}
